### PR TITLE
Asana source + tweak converters to adapt to it

### PR DESCRIPTION
**Pull request by ypc-faros**

### CODE REVIEW

File: sources/asana-source/src/streams/tasks.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  get cursorField(): string | string[] {
+    return 'modified_at';
+  }
+
+  async *streamSlices(): AsyncGenerator<StreamSlice> {
+    const asana = Asana.instance(this.config);
+
+    for (const workspace of await asana.getWorkspaces()) {
+      yield {workspace: workspace.gid};
```
Comment (willmarks) [inline sources/asana-source/src/streams/tasks.ts:39]: What is a workspace in the context of Asana again? Is it like our 'FAI' in Jira?

**Review by willmarks — APPROVED** [review]
Nice work!

File: destinations/airbyte-faros-destination/src/converters/asana/tasks.ts [hunk 2]
```diff
@@ -29,14 +29,34 @@ interface TmsTaskStatus {
...
+  private _config: Config = undefined;
+
+  private initialize(ctx?: StreamContext): void {
+    this._config =
+      this._config ?? ctx?.config.source_specific_configs?.asana ?? {};
+  }
+
+  private get config(): Config {
+    return this._config;
+  }
```
[the remark this answers is not in the document]
Comment (ypc-faros) [reply destinations/airbyte-faros-destination/src/converters/asana/tasks.ts:58]: Yeah it doesn't bring any value here. I was fiddling with a refactor earlier and this chunk survived the copy/paste/modify session 🗡️ 

I'll remove

File: destinations/airbyte-faros-destination/src/converters/asana/tasks.ts [hunk 3]
```diff
@@ -77,14 +120,13 @@ export class Tasks extends AsanaConverter {
         ),
         url: task.permalink_url ?? null,
         type: AsanaCommon.toTmsTaskType(task.resource_type),
-        priority: typeof priority === 'string' ? priority : null,
         status: this.getStatus(task),
-        points: typeof points === 'number' ? points : null,
```
[the remark this answers is not in the document]
Comment (ypc-faros) [reply destinations/airbyte-faros-destination/src/converters/asana/tasks.ts:82]: Not for now. We'll probably have to add a configuration field to tell us which custom field to use. But there's no immediate need for these now.

File: sources/asana-source/resources/spec.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+              },
+              "personal_access_token": {
+                "type": "string",
+                "title": "Personal Access Token",
+                "description": "Asana Personal Access Token (generate yours <a href=\"https://app.asana.com/0/developer-console\">here</a>).",
+                "airbyte_secret": true
+              }
+            }
+          }
+        ]
```
[the remark this answers is not in the document]
Comment (ypc-faros) [reply sources/asana-source/resources/spec.json:35]: Yeah exactly. The community source actually supports oauth. We might need it later

File: sources/asana-source/resources/spec.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        "description": "The number of records to request from the API per call.",
+        "default": 100
+      },
+      "api_timeout": {
+        "type": "integer",
+        "title": "API Timeout",
+        "description": "Timeout (in milliseconds) to use when making requests to Asana API. 0 means no timeout.",
+        "default": 0
+      },
+      "start_date": {
```
[the remark this answers is not in the document]
Comment (ypc-faros) [reply sources/asana-source/resources/spec.json:61]: I like to have the possibility to run on an arbitrary time range. But this is a good point. I think we could work on a common way to accept time ranges and use them across the board.

File: sources/asana-source/src/asana.ts [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+  Story,
+  Tag,
+  Task,
+  User,
+  Workspace,
+} from './models';
+
+export const MIN_DATE = new Date(0).toISOString();
+// January 1, 2200
+export const MAX_DATE = new Date(7258118400000).toISOString();
```
[the remark this answers is not in the document]
Comment (ypc-faros) [reply sources/asana-source/src/asana.ts:21]: I'm always a little paranoid with boundaries and prefer to play it safe :P
I cannot come up with an issue with using `now` I'll think about it :D

File: sources/asana-source/src/streams/tasks.ts (hunk 1 above)
Comment (ypc-faros) [reply sources/asana-source/src/streams/tasks.ts:39]: I don't understand it completely but it groups tasks, users, projects, etc. I think users can have multiple workspaces and organizations (I don't know the difference). Most of the api's need a workspace id

**Pull request merged**